kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs

```diff
@@ -289,11 +282,15 @@ impl CodegenBackend for GotocCodegenBackend {
                 }
                 ReachabilityType::None => {}
                 ReachabilityType::PubFns => {
-                    let entry_fn = tcx.entry_fn(()).map(|(id, _)| id);
-                    let local_reachable = filter_crate_items(tcx, |_, def_id| {
-                        (tcx.is_reachable_non_generic(def_id) && tcx.def_kind(def_id).is_fn_like())
-                            || entry_fn == Some(def_id)
-                    });
+                    let main_instance =
+                        stable_mir::entry_fn().map(|main_fn| Instance::try_from(main_fn).unwrap());
+                    let local_reachable = filter_crate_items(tcx, |_, instance| {
+                        let def_id = rustc_internal::internal(instance.def.def_id());
+                        Some(instance) == main_instance || tcx.is_reachable_non_generic(def_id)
+                    })
+                    .into_iter()
+                    .map(MonoItem::Fn)
```

Comment:
Oh, looks like it's not only the harnesses, but any reachable function. Now I really want to know why 😅

Reply:
No, it's just the harness